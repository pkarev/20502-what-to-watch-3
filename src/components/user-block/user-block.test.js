import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';
import UserBlock from './user-block.jsx';
import NameSpace from '../../reducer/name-space';
import history from '../../history';

const mockStore = configureStore([]);

describe(`Reder UserBlock`, () => {

  it(`Render sign in link if not authorized`, () => {
    const store = mockStore({
      [NameSpace.USER]: {
        isAuthorized: false,
      }
    });

    const tree = renderer
      .create(
          <Router history={history}>
            <Provider store={store}>
              <UserBlock/>
            </Provider>
          </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render user avatar if authorized`, () => {
    const store = mockStore({
      [NameSpace.USER]: {
        isAuthorized: true,
      }
    });

    const tree = renderer
    .create(
        <Router history={history}>
          <Provider store={store}>
            <UserBlock/>
          </Provider>
        </Router>
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

