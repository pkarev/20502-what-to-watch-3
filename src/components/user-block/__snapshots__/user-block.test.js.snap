// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reder UserBlock Render sign in link if not authorized 1`] = `
<div
  className="user-block"
>
  <a
    href="#"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;

exports[`Reder UserBlock Render user avatar if authorized 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <img
      alt="User avatar"
      height="63"
      src="img/avatar.jpg"
      width="63"
    />
  </div>
</div>
`;
